File: api-tester/src/main/java/com/revenuecat/apitester/java/PricingPhaseAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.OfferPaymentMode;
import com.revenuecat.purchases.models.Period;
import com.revenuecat.purchases.models.Price;
import com.revenuecat.purchases.models.PricingPhase;
import com.revenuecat.purchases.models.RecurrenceMode;

import java.util.Locale;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class PricingPhaseAPI {
    static void checkPricingPhase(PricingPhase pricingPhase) {
        final Locale locale = Locale.getDefault();
        Period billingPeriod = pricingPhase.getBillingPeriod();
        RecurrenceMode recurrenceMode = pricingPhase.getRecurrenceMode();
        Integer billingCycleCount = pricingPhase.getBillingCycleCount();
        Price price = pricingPhase.getPrice();
        String pricePerMonthString = pricingPhase.formattedPriceInMonths(locale);
        String pricePerMonthStringNoLocale = pricingPhase.formattedPriceInMonths();
        Price pricePerWeek = pricingPhase.pricePerWeek(locale);
        Price pricePerMonth = pricingPhase.pricePerMonth(locale);
        Price pricePerYear = pricingPhase.pricePerYear(locale);
        Price pricePerWeekNoLocale = pricingPhase.pricePerWeek();
        Price pricePerMonthNoLocale = pricingPhase.pricePerMonth();
        Price pricePerYearNoLocale = pricingPhase.pricePerYear();

        OfferPaymentMode offerPaymentMode = pricingPhase.getOfferPaymentMode();
    }

    static void checkPrice(Price price) {
        String formatted = price.getFormatted();
        Long amount = price.getAmountMicros();
        String currencyCode = price.getCurrencyCode();
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PricingPhaseAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.OfferPaymentMode
import com.revenuecat.purchases.models.Period
import com.revenuecat.purchases.models.Price
import com.revenuecat.purchases.models.PricingPhase
import com.revenuecat.purchases.models.RecurrenceMode
import java.util.Locale

@Suppress("unused", "UNUSED_VARIABLE", "RemoveExplicitTypeArguments")
private class PricingPhaseAPI {
    fun checkPricingPhase(pricingPhase: PricingPhase, locale: Locale) {
        val billingPeriod: Period = pricingPhase.billingPeriod
        val recurrenceMode: RecurrenceMode = pricingPhase.recurrenceMode
        val billingCycleCount: Int? = pricingPhase.billingCycleCount
        val price: Price = pricingPhase.price
        val pricePerMonthString: String = pricingPhase.formattedPriceInMonths(locale)
        val pricePerMonthStringNoLocale: String = pricingPhase.formattedPriceInMonths()
        val pricePerDay = pricingPhase.pricePerDay(locale)
        val pricePerWeek = pricingPhase.pricePerWeek(locale)
        val pricePerMonth = pricingPhase.pricePerMonth(locale)
        val pricePerYear = pricingPhase.pricePerYear(locale)
        val pricePerDayNoLocale = pricingPhase.pricePerDay()
        val pricePerWeekNoLocale = pricingPhase.pricePerWeek()
        val pricePerMonthNoLocale = pricingPhase.pricePerMonth()
        val pricePerYearNoLocale = pricingPhase.pricePerYear()

        val offerPaymentMode: OfferPaymentMode? = pricingPhase.offerPaymentMode
    }

    fun checkingPrice(price: Price) {
        val formatted: String = price.formatted
        val amountMicros: Long = price.amountMicros
        val currencyCode: String = price.currencyCode
    }
}
